src/test/java/org/schemaspy/output/dot/schemaspy/DotFileTest.java

```diff
@@ -0,0 +1,42 @@
+package org.schemaspy.output.dot.schemaspy;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.schemaspy.output.dot.DotException;
+import org.schemaspy.output.dot.schemaspy.graph.EmptyGraph;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class DotFileTest {
+
+    @TempDir
+    private Path outputDir;
+
+    @Test
+    void returnsFalseWhenEmptyGraphReturnedFromGraphProducer() {
```

Comment:
This and many other tests (I won't comment on the rest) is on far too low level of abstraction.

Reply:
This specific scenario will disappear, but could you elaborate on what you mean by
> is on far too low level of abstraction.

Reply:
I mean that no part of the test expresses the intent in [human terms](https://martinfowler.com/bliki/UbiquitousLanguage.html). As an aside, note that we found the design of `write` to be sub-optimal. That is not a coincidence, in most cases.